src/sa_web/views.py

```diff
@@ -361,12 +361,26 @@ def api(request, path):
     A small proxy for a Shareabouts API server, exposing only
     one configured dataset.
     """
-    root = settings.SHAREABOUTS.get('DATASET_ROOT')
+    
+    #root = settings.SHAREABOUTS.get('DATASET_ROOT')
+    root = request.GET.get('url')
+    dataset_id = request.GET.get('id')
+    print("root", root)
+    print("dataset_id", dataset_id)
+
+    # Clear out request object query string dictionary. For some reason, /api/places 
+    # will return an empty dataset if a query string is included in the request
+    request.GET = request.GET.copy()
+    if 'url' in request.GET:
+        request.GET.pop('url')
+    if 'id' in request.GET:
+        request.GET.pop('id')
```

Comment:
I think the query params aren't the cleanest implementation, plus it may interfere in other situations where we may actually want to pass query params to the API. I think we can clean this up by creating a new route that contains only the dataset id. I believe that we can create a new endpoint, like `/dataset/<datasetId>/places`, by changing our current `api_proxy` line in our `urls.py` to this two lines:

```
    url(r'^dataset/(?P<dataset_id>[^/]+)/(?P<path>.*)$', views.api, name='dataset_api_proxy'),
```
and change the `views.api` method definition to this:
```
def api(request, path, **kwargs):
```
Then we can check whether we have a `dataset_id` variable passed by checking `kwargs['dataset_id']`.

Which will forward a call to a url like `/dataset/trees/places` to our `views.api` function, with `path=places` and `dataset_id=trees`. Then, we can use `dataset_id` to look up our url and dataset key environment variables - in this case, `TREES_SITE_URL` and `TREES_DATASET_KEY`.

Check this out for a reference on Django's urls: https://docs.djangoproject.com/en/1.7/topics/http/urls/#named-groups